src/features/regexp-v-flag.md

```diff
@@ -0,0 +1,252 @@
+---
+title: 'RegExp `v` flag with set notation and properties of strings'
+author: 'Mark Davis ([@mark_e_davis](https://twitter.com/mark_e_davis)), Markus Scherer, and Mathias Bynens ([@mathias](https://twitter.com/mathias))'
+avatars:
+  - 'mark-davis'
+  - 'markus-scherer'
+  - 'mathias-bynens'
+date: 2022-06-15
+tags:
+  - ECMAScript
+description: 'The new RegExp `v` flag enables `unicodeSets` mode, unlocking support for extended character classes, including Unicode properties of strings, set notation, and improved case-insensitive matching.'
+---
+JavaScript has supported regular expressions since ECMAScript 3 (1999). Sixteen years later, ES2015 introduced [Unicode mode (the `u` flag)](https://mathiasbynens.be/notes/es6-unicode-regex), [sticky mode (the `y` flag)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/sticky#description), and [the `RegExp.prototype.flags` getter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/flags). Another three years later, ES2018 introduced [`dotAll` mode (the `s` flag)](https://mathiasbynens.be/notes/es-regexp-proposals#dotAll), [lookbehind assertions](https://mathiasbynens.be/notes/es-regexp-proposals#lookbehinds), [named capture groups](https://mathiasbynens.be/notes/es-regexp-proposals#named-capture-groups), and [Unicode character property escapes](https://mathiasbynens.be/notes/es-unicode-property-escapes). And in ES2020, [`String.prototype.matchAll`](https://v8.dev/features/string-matchall) made it easier to work with regular expressions. JavaScript regular expressions have come a long way, and are still improving.
+
+The latest example of this is [the new `unicodeSets` mode, enabled using the `v` flag](https://github.com/tc39/proposal-regexp-v-flag). This new mode unlocks support for _extended character classes_, including the following features:
+
+- [Unicode properties of strings](/features/regexp-v-flag#unicode-properties-of-strings)
+- [set notation + string literal syntax](/features/regexp-v-flag#set-notation)
+- [improved case-insensitive matching](/features/regexp-v-flag#ignoreCase)
+
+This article dives into each of these. But first things first — here’s how to use the new flag:
+
+```js
+const re = /…/v;
+```
+
+The `v` flag can be combined with existing regular expression flags, with one notable exception. The `v` flag enables all the good parts of the `u` flag, but with additional features and improvements. Crucially, `v` is a completely separate mode from `u` rather than a complementary one. For this reason, the `v` and `u` flags cannot be combined — trying to use both flags on the same regular expression results in an error. The only valid options are: either use `u`, or use `v`, or use neither `u` nor `v`. But since `v` is the most feature-complete option, that choice is easily made…
```

Comment:
The "The `v` flag enables all the good parts of the `u` flag, but with additional features and improvements." sentence makes it sound like `v` is a strict feature superset of `u`.

Maybe "The `v` flag enables all the good parts of the `u` flag, albeit with additional incompatible features and improvements."?

Reply:
Or

> The `v` flag enables all the good parts of the `u` flag, but with additional features and improvements — some of which are backwards-incompatible with the `u` flag.

Reply:
Agree with @mathiasbynens

Reply:
```suggestion
The `v` flag can be combined with existing regular expression flags, with one notable exception. The `v` flag enables all the good parts of the `u` flag, but with additional features and improvements — some of which are backwards-incompatible with the `u` flag. Crucially, `v` is a completely separate mode from `u` rather than a complementary one. For this reason, the `v` and `u` flags cannot be combined — trying to use both flags on the same regular expression results in an error. The only valid options are: either use `u`, or use `v`, or use neither `u` nor `v`. But since `v` is the most feature-complete option, that choice is easily made…
```

Reply:
Suggestion applied. Thanks for pointing this out, @syg!